pennylane/qnodes/cv.py

```diff
@@ -81,6 +81,11 @@ def _best_method(self, idx):
                     elif ob.return_type == ObservableReturnTypes.Variance:
                         if ob.ev_order is None or ob.ev_order >= 2:
                             x = "F"
+
+                    elif ob.return_type is ObservableReturnTypes.Probability:
+                        raise ValueError(
+                            "Analytic gradients are not supported for circuits extracting Fock state probabilities."
+                        )

```

Comment:
@antalszava, I think this should be setting `x = "F"` rather than raising an error.

This method returns the _allowed_ diff methods for each parameters, and the validation happens instead in `JacobianQNode.jacobian()`

Reply:
I believe - if I'm wrong let me know!

Reply:
Oh, that's a good point! thanks for catching that! Moved it to the jacobian